src/pages/batchImports/[resourceId]/index.js

```diff
@@ -0,0 +1,296 @@
+
+  const missingFields = gridData.list.flatMap(row =>
+    mandatoryColumns
+      .filter(col => row[col.field] === null || row[col.field] === undefined || row[col.field] === '')
+      .map(col => col.headerName)
+  )
+
+  if (missingFields.length > 0) {
+    const uniqueMissingFields = [...new Set(missingFields)]
+    stackError({
+      message: `${uniqueMissingFields.join(', ')} ${uniqueMissingFields.length > 1 ? 'are' : 'is'} mandatory field${
+        uniqueMissingFields.length > 1 ? 's' : ''
+      }.`
+    })
+
+    return
+  }
+
+  const convertedData = gridData.list.map(row => {
+    return Object.keys(row).reduce((acc, key) => {
+      const col = columns.find(c => c.field === key)
+      let value = row[key]
+      value = value === '' ? null : value
+      acc[key] = col ? convertValue(value, col.dataType, true) : value
+
+      return acc
+    }, {})
+  })
+
+  return convertedData
+}
+
+const BatchImports = () => {
+  const { stack } = useWindow()
+  const { stack: stackError } = useError()
+  const router = useRouter()
+
+  const [columns, setColumns] = useState([])
+  const [gridData, setGridData] = useState([])
+  const [name, setName] = useState('')
+  const [objectName, setObjectName] = useState('')
+  const [endPoint, setEndPoint] = useState('')
+
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { resourceId } = router.query
+  const { platformLabels } = useContext(ControlContext)
+
+  const { access } = useResourceQuery({
+    datasetId: resourceId
+  })
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (resourceId) {
+          const res = await getRequest({
+            extension: SystemRepository.ETL.get,
+            parameters: `_resourceId=${resourceId}`
+          })
+
+          const modifiedFields = res.record.fields.map(({ name, dataType, ...rest }) => ({
+            field: name,
+            headerName: name,
+            type: (dataType == 2 || dataType == 3) && 'number',
+            flex: 1,
+            ...rest
+          }))
+
+          setColumns([{ field: 'recordId', headerName: '', flex: 0.4 }, ...modifiedFields])
+
+          setObjectName(res.record.objectName)
+          setEndPoint(res.record.endPoint)
+        }
+      } catch (exception) {}
+    })()
+  }, [])
+
+  const refetch = () => {
+    setGridData(prevGridData => {
+      const transformedData = transform(prevGridData.list)
+
+      return transformedData
+    })
+  }
+
+  const handleFileChange = event => {
+    const file = event.target.files[0]
+
+    setName(file.name)
+
+    if (file) {
+      const reader = new FileReader()
+      reader.onload = e => {
+        const text = e.target.result
+        const data = parseCSV(text, columns)
+        setGridData(data)
+      }
+      reader.readAsText(file)
+    }
+  }
+
+  const clearFile = () => {
+    setName('')
+    setGridData([])
+    document.getElementById('csvInput').value = null
```

Comment:
csvInput is an input field. current csvInput value should be a state. Clear the state here and the component should know to make it empty.